src/app/@ansyn/map-facade/components/imagery-status/imagery-status.component.ts

```diff
@@ -48,45 +50,61 @@ import { get as _get } from 'lodash';
 	init: 'ngOnInit',
 	destroy: 'ngOnDestroy'
 })
-export class ImageryStatusComponent implements OnInit, OnDestroy {
-	_overlay: any;
-	selectedMap = 'openLayersMap';
+export class ImageryStatusComponent implements OnInit, OnDestroy, IEntryComponent {
+	// @todo refactor
+	overlay: any;
+	_mapState: IMapSettings;
+	mapsAmount = 1;
+	@HostBinding('class.active') isActiveMap: boolean;
+
+	get selectedMap() {
+		return _get(this.communicators.provide(this.mapState.id), 'ActiveMap.mapType');
+	}
+
+	@Input()
+	set mapState(value: IMapSettings) {
+		if (_get(this._mapState, 'data.overlay') !== _get(value, 'data.overlay')) {
```

Comment:
i don't like the lodash way, can you find a better way?

Reply:
lodash `get` is great 🥇